### PR TITLE
[FLINK-16076] Translate "Queryable State" page into Chinese

### PR DESCRIPTION
**Pull request by banmoy**
## What is the purpose of the change

FLINK-16076 Translate "Queryable State" page into Chinese

## Brief change log

Translate "Queryable State" page into Chinese

## Verifying this change

No need for test.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by banmoy** [comment]
@klion26 Can you help to review this translation? Thanks.

File: docs/dev/stream/state/queryable_state.zh.md [hunk 1]
```diff
@@ -202,35 +174,29 @@ CompletableFuture<S> getKvState(
     StateDescriptor<S, V> stateDescriptor)
 {% endhighlight %}
 
-The above returns a `CompletableFuture` eventually holding the state value for the queryable state instance identified 
-by `queryableStateName` of the job with ID `jobID`. The `key` is the key whose state you are interested in and the 
-`keyTypeInfo` will tell Flink how to serialize/deserialize it. Finally, the `stateDescriptor` contains the necessary 
-information about the requested state, namely its type (`Value`, `Reduce`, etc) and the necessary information on how 
-to serialize/deserialize it.
+该方法会返回一个`CompletableFuture`，最终会保存作业ID为`jobID`，可查询状态名称为`queryableStateName`的状态。`key`指定所要查询的状态所属的key，
```
Comment (carp84) [inline docs/dev/stream/state/queryable_state.zh.md:177]: 该方法会返回一个最终将包含状态值的queryable state实例，该实例可通过`JobID`和`queryableStateName`识别。在方法参数中，`key`用来指定所要查询的状态所属的key

File: docs/dev/stream/state/queryable_state.zh.md [hunk 2]
```diff
@@ -288,34 +254,27 @@ resultFuture.thenAccept(response -> {
...
-default as they would not be exposed anywhere. As soon as there is better support to publish these
-numbers via the Metrics system, we should enable the stats.
+* `queryable-state.proxy.ports`: 代理的服务端口范围。 如果同一台机器上运行了多个task manager，可以避免端口冲突。指定的范围可以是一个具体的端口号，如"9123"，
+   可以是一个端口范围，如"50100-50200"，或者可以是端口范围以及端口号的组合，如"50100-50200,50300-50400,51234"。默认端口号是9069。
+* `queryable-state.proxy.network-threads`: 代理上network (event loop) thread的数量，用来接收查询请求 (0 => #slots)。
+* `queryable-state.proxy.query-threads`: 代理上处理查询请求的线程数 (0 => #slots)。
+
+## 限制
+
+* 可查询状态的生命周期受限于作业的生命周期，*比如，* tasks在启动是注册可查询状态，并在销毁时注销。在未来的版本中，希望能够将其解耦
```
Comment (carp84) [inline docs/dev/stream/state/queryable_state.zh.md:274]: *比如，* -> *比如*
启动是 -> 启动时

File: docs/dev/stream/state/queryable_state.zh.md [hunk 3]
```diff
@@ -288,34 +254,27 @@ resultFuture.thenAccept(response -> {
...
 
 ### State Server
-* `queryable-state.server.ports`: the server port range of the queryable state server. This is useful to avoid port clashes if more 
-   than 1 task managers run on the same machine. The specified range can be: a port: "9123", a range of ports: "50100-50200",
-   or a list of ranges and or points: "50100-50200,50300-50400,51234". The default port is 9067.
-* `queryable-state.server.network-threads`: number of network (event loop) threads receiving incoming requests for the state server (0 => #slots)
-* `queryable-state.server.query-threads`: number of threads handling/serving incoming requests for the state server (0 => #slots).
+* `queryable-state.server.ports`: 服务器端口范围。如果同一台机器上运行了多个task manager，可以避免端口冲突。指定的范围可以是一个具体的端口号，如"9123"，
+    可以是一个端口范围，如"50100-50200"，或者可以是端口范围以及端口号的组合，如"50100-50200,50300-50400,51234"。默认端口号是9067。
+* `queryable-state.server.network-threads`: 服务器端network (event loop) thread的数量，用来接收查询请求 (0 => #slots)。
```
[the remark this answers is not in the document]
Comment (klion26) [reply docs/dev/stream/state/queryable_state.zh.md:262]: 这里的 ”0 => #slots“ 是不是应该按照 javaDoc 进行翻译一下，也就是设置为 0 的使用，实际使用 #slots 个线程

File: docs/dev/stream/state/queryable_state.zh.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (klion26) [reply docs/dev/stream/state/queryable_state.zh.md:274]: ”销毁时“换成”退出时“会好一些吗？

File: docs/dev/stream/state/queryable_state.zh.md [hunk 4]
```diff
@@ -288,34 +254,27 @@ resultFuture.thenAccept(response -> {
...
+   可以是一个端口范围，如"50100-50200"，或者可以是端口范围以及端口号的组合，如"50100-50200,50300-50400,51234"。默认端口号是9069。
+* `queryable-state.proxy.network-threads`: 代理上network (event loop) thread的数量，用来接收查询请求 (0 => #slots)。
+* `queryable-state.proxy.query-threads`: 代理上处理查询请求的线程数 (0 => #slots)。
+
+## 限制
+
+* 可查询状态的生命周期受限于作业的生命周期，*比如，* tasks在启动是注册可查询状态，并在销毁时注销。在未来的版本中，希望能够将其解耦
+从而允许task结束后能够查询状态，并且通过状态备份来加速恢复。
+* 现在是通过简单的tell来通知可用的KvState。将来会使用asks和acknowledgements来提升稳定性。
+* 服务器端和客户端会记录请求的统计信息。这个功能默认没有开启，因为统计信息目前不会暴露给外部。如果将来支持通过Metrics系统发布这些数据，统计功能
```
[the remark this answers is not in the document]
Comment (klion26) [reply docs/dev/stream/state/queryable_state.zh.md:277]: ”将开启统计功能“ 会更好一些吗？

**Comment by banmoy** [comment]
@flinkbot run travis

File: docs/dev/stream/state/queryable_state.zh.md [hunk 5]
```diff
@@ -27,75 +27,54 @@ under the License.
...
-  <strong>Attention:</strong> When querying a state object, that object is accessed from a concurrent
-  thread without any synchronization or copying. This is a design choice, as any of the above would lead
-  to increased job latency, which we wanted to avoid. Since any state backend using Java heap space,
-  <i>e.g.</i> <code>MemoryStateBackend</code> or <code>FsStateBackend</code>, does not work
-  with copies when retrieving values but instead directly references the stored values, read-modify-write
-  patterns are unsafe and may cause the queryable state server to fail due to concurrent modifications.
-  The <code>RocksDBStateBackend</code> is safe from these issues.
+  <strong>注意:</strong> 进行查询时，state 会在并发线程中被访问，但 state 不会进行同步和拷贝。这种设计是为了避免同步和拷贝带来的作业延时。对于使用 Java 堆内存的 state backend，
+  <i>比如</i> <code>MemoryStateBackend</code> 或者 <code>FsStateBackend</code>，它们获取状态时不会进行拷贝，而是直接引用状态对象，所以对状态的 read-modify-write 是不安全的，并且
+ 可能会因为并发修改导致查询失败。但 <code>RocksDBStateBackend</code> 是安全的，不会遇到上述问题。
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]: 这里不需要空行，空行会多加一个空格

File: docs/dev/stream/state/queryable_state.zh.md [hunk 6]
```diff
@@ -27,75 +27,54 @@ under the License.
...
+ 2. `QueryableStateClientProxy`，运行在每个 `TaskManager` 上(*即* Flink 集群内部)，负责接收客户端的查询请求，从所负责的 Task Manager 获取请求的 state，并返回给客户端；
+ 3. `QueryableStateServer`, 运行在 `TaskManager` 上，负责服务本地存储的 state。
 
-The client connects to one of the proxies and sends a request for the state associated with a specific
-key, `k`. As stated in [Working with State]({{ site.baseurl }}/dev/stream/state/state.html), keyed state is organized in
-*Key Groups*, and each `TaskManager` is assigned a number of these key groups. To discover which `TaskManager` is
-responsible for the key group holding `k`, the proxy will ask the `JobManager`. Based on the answer, the proxy will
-then query the `QueryableStateServer` running on that `TaskManager` for the state associated with `k`, and forward the
-response back to the client.
+客户端连接到一个代理，并发送请求获取特定 `k` 对应的 state。 如 [Working with State]({{ site.baseurl }}/dev/stream/state/state.html)所述，keyed state 按照
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]:
```suggestion
客户端连接到一个代理，并发送请求获取特定 `k` 对应的 state。 如 [Working with State]({{ site.baseurl }}/zh/dev/stream/state/state.html) 所述，keyed state 按照
```

File: docs/dev/stream/state/queryable_state.zh.md [hunk 7]
```diff
@@ -27,75 +27,54 @@ under the License.
...
-from the `opt/` folder of your [Flink distribution](https://flink.apache.org/downloads.html "Apache Flink: Downloads"),
-to the `lib/` folder.
- 2. set the property `queryable-state.enable` to `true`. See the [Configuration]({{ site.baseurl }}/ops/config.html#queryable-state) documentation for details and additional parameters.
+ 1. 将 `flink-queryable-state-runtime{{ site.scala_version_suffix }}-{{site.version }}.jar`
+从 [Flink distribution](https://flink.apache.org/downloads.html "Apache Flink: Downloads") 的 `opt/` 目录拷贝到 `lib/` 目录；
+ 2. 将参数 `queryable-state.enable` 设置为 `true`。详细信息以及其它配置可参考文档 [Configuration]({{ site.baseurl }}/ops/config.html#queryable-state)。
 
-To verify that your cluster is running with queryable state enabled, check the logs of any 
-task manager for the line: `"Started the Queryable State Proxy Server @ ..."`.
+为了验证集群的 queryable stat 已经被激活，可以检查任意 task manager 的日志中是否包含 "Started the Queryable State Proxy Server @ ..."。
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]:
```suggestion
为了验证集群的 queryable state 已经被激活，可以检查任意 task manager 的日志中是否包含 "Started the Queryable State Proxy Server @ ..."。
```

File: docs/dev/stream/state/queryable_state.zh.md [hunk 8]
```diff
@@ -150,20 +126,18 @@ descriptor.setQueryable("query-name"); // queryable state name
...
-## Querying State
+## 查询 state
 
-So far, you have set up your cluster to run with queryable state and you have declared (some of) your state as
-queryable. Now it is time to see how to query this state. 
+目前为止，你已经激活了集群的 queryable state 功能，并且将一些 state 设置成了可查询的，接下来将会展示如何进行查询。
 
-For this you can use the `QueryableStateClient` helper class. This is available in the `flink-queryable-state-client` 
-jar which must be explicitly included as a dependency in the `pom.xml` of your project along with `flink-core`, as shown below:
+为了进行查询，可以使用辅助类 `QueryableStateClient`，这个类位于 `flink-queryable-state-client` 的jar中，在项目的 `pom.xml` 需要显示添加
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]:
```suggestion
为了进行查询，可以使用辅助类 `QueryableStateClient`，这个类位于 `flink-queryable-state-client` 的 jar 中，在项目的 `pom.xml` 需要显示添加
```

File: docs/dev/stream/state/queryable_state.zh.md [hunk 9]
```diff
@@ -180,18 +154,16 @@ jar which must be explicitly included as a dependency in the `pom.xml` of your p
 {% endhighlight %}
 </div>
 
-For more on this, you can check how to [set up a Flink program]({{ site.baseurl }}/dev/projectsetup/dependencies.html).
+关于依赖的更多信息, 可以参考如何[配置Flink项目]({{ site.baseurl }}/zh/dev/projectsetup/dependencies.html).
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]:
```suggestion
关于依赖的更多信息, 可以参考如何[配置 Flink 项目]({{ site.baseurl }}/zh/dev/projectsetup/dependencies.html).
```

File: docs/dev/stream/state/queryable_state.zh.md [hunk 10]
```diff
@@ -27,75 +27,54 @@ under the License.
...
 
-The client connects to one of the proxies and sends a request for the state associated with a specific
-key, `k`. As stated in [Working with State]({{ site.baseurl }}/dev/stream/state/state.html), keyed state is organized in
-*Key Groups*, and each `TaskManager` is assigned a number of these key groups. To discover which `TaskManager` is
-responsible for the key group holding `k`, the proxy will ask the `JobManager`. Based on the answer, the proxy will
-then query the `QueryableStateServer` running on that `TaskManager` for the state associated with `k`, and forward the
-response back to the client.
+客户端连接到一个代理，并发送请求获取特定 `k` 对应的 state。 如 [Working with State]({{ site.baseurl }}/dev/stream/state/state.html)所述，keyed state 按照
+*Key Groups* 进行划分，每个 `TaskManager` 会分配其中的一些 key groups。代理会询问 `JobManager` 以找到 `k` 所属 key group 的 TaskManager。根据返回的结果, 代理
+将会向运行在 `TaskManager` 上的 `QueryableStateServer` 查询 `k` 对应的 state， 并将结果返回给客户端。
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]: 这里不需要换行

File: docs/dev/stream/state/queryable_state.zh.md [hunk 11]
```diff
@@ -119,28 +98,25 @@ QueryableStateStream asQueryableState(
...
 </div>
 
-The returned `QueryableStateStream` can be seen as a sink and **cannot** be further transformed. Internally, a 
-`QueryableStateStream` gets translated to an operator which uses all incoming records to update the queryable state 
-instance. The updating logic is implied by the type of the `StateDescriptor` provided in the `asQueryableState` call. 
-In a program like the following, all records of the keyed stream will be used to update the state instance via the 
-`ValueState.update(value)`:
+返回的 `QueryableStateStream` 可以被视作一个sink，而且**不能再**被进一步转换。在内部实现上，一个 `QueryableStateStream` 被转换成一个 operator，
+使用输入的数据来更新 queryable state。state 如何更新是由 `asQueryableState` 提供的 `StateDescriptor` 来决定的。在下面的代码中, keyed stream 的所有数据
+将会通过 `ValueState.update(value)` 来更新状态：
```
Comment (klion26) [inline docs/dev/stream/state/queryable_state.zh.md]: 这里不需要换行

**Pull request closed without merging**